engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -536,16 +536,18 @@ public static FileSystem newFileSystemInstance(Configuration conf) throws IOExce
    * @param dfsRootDir  value of dt.dfsRootDir or apex.app.dfsRootDir
    * @param userShortName  current user short name (either login user or current user depending on impersonation settings)
    * @param prependHomeDir  prepend user's home dir if dfsRootDir is relative path
-
+   * @param updateConf update the conf object with the evaluated value
    * @return
    */
   private static Path evalDFSRootDir(FileSystem fs, Configuration conf, String dfsRootDir, String userShortName,
-      boolean prependHomeDir)
+      boolean prependHomeDir, boolean updateConf)
   {
     try {
       if (userShortName != null && dfsRootDir.contains(DT_DFS_USER_NAME)) {
         dfsRootDir = dfsRootDir.replace(DT_DFS_USER_NAME, userShortName);
-        conf.set(DT_DFS_ROOT_DIR, dfsRootDir);
+        if (updateConf) {
+          conf.set(DT_DFS_ROOT_DIR, dfsRootDir);
```

Comment:
@PramodSSImmaneni  As mentioned elsewhere, we cannot do the same thing for APEX_DFS_ROOT_DIR since the value is per Hadoop login user so cannot cache it in the conf object